apps/api/src/app/workflows-v2/usecases/validate-control-values/validate-control-values-and-add-defaults.usecase.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { ContentIssue, JSONSchemaDto, StepContentIssueEnum } from '@novu/shared';
+import { ExtractDefaultsUsecase } from '../get-default-values-from-schema/extract-defaults.usecase';
+import { ValidateControlValuesCommand } from './validate-control-values.command';
+import { ValidateControlValuesResponse } from './validate-control-values.response';
+import { findMissingKeys } from '../../util/usecaseutils';
+import _ = require('lodash');
+
+@Injectable()
+export class ValidateControlValuesAndAddDefaultsUseCase {
+  constructor(private extractDefaultsUseCase: ExtractDefaultsUsecase) {}
+
+  execute(command: ValidateControlValuesCommand): ValidateControlValuesResponse {
+    const defaultValues = this.extractDefaultsUseCase.execute({
+      jsonSchemaDto: command.controlSchema.schema as JSONSchemaDto,
+    });
+
+    return {
+      augmentedControlValues: _.merge(defaultValues, command.controlValues),
+      issuesMissingValues: this.buildMissingControlValuesIssuesList(defaultValues, command.controlValues),
+    };
+  }
+
+  private buildMissingControlValuesIssuesList(defaultValues: Record<string, any>, controlValues: Record<string, any>) {
+    const missingRequiredControlValues = findMissingKeys(defaultValues, controlValues);
+
+    return this.buildControlPreviewIssues(missingRequiredControlValues);
+  }
+
+  private buildControlPreviewIssues(keys: string[]): Record<string, ContentIssue[]> {
```

Comment:
Following the renaming in earlier files...
```suggestion
  private buildControIssues(keys: string[]): Record<string, ContentIssue[]> {
```